tests/unit/symbolic_regression/test_explicit_regression.py

```diff
@@ -310,3 +312,10 @@ def test_explicit_regression_relative_get_fit_vec_and_jac(
 
 def test_can_pickle(sample_regression):
     _ = dill.loads(dill.dumps(sample_regression))
+
+
+def test_can_copy(sample_regression):
```

Comment:
does this test automatically test the desructors you implemented?

Reply:
It tests the issue that was causing the segmentation fault before, which was solved by changing the destructors. But this test doesn't test for any memory leaks/that the training data was deleted correctly if that answers your question.

Reply:
yep, thats clear now, thanks

Reply:
of course no problem